### PR TITLE
feat: add fallback dependency cache keys [TSD-1141]

### PR DESCRIPTION
**Pull request by emann**

### CODE REVIEW

File: .github/workflows/python-deploy-serverless.yml [hunk 1]
```diff
@@ -45,11 +45,12 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Setup Python and Poetry and Install Dependencies
-        uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@v2
+        uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@TSD-1141-add-fallback-dependency-cache-keys
```
Comment (sweetb) [inline .github/workflows/python-deploy-serverless.yml]: Do we want this using a specific branch?

File: .github/workflows/python-publish-package.yml [hunk 2]
```diff
@@ -35,17 +35,12 @@ jobs:
...
 
-    - name: Cache dependencies
-      uses: actions/cache@v2
-      with:
-        path: ./.venv
-        key: poetry-${{ inputs.dependency-hash-version }}-${{ hashFiles('poetry.lock') }}
-
     - name: Setup Python and Poetry and Install Dependencies
-      uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@v2
+      uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@TSD-1141-add-fallback-dependency-cache-keys
```
Comment (sweetb) [inline .github/workflows/python-publish-package.yml]: Same

File: .github/workflows/python-run-checks-and-tests.yml [hunk 3]
```diff
@@ -62,19 +57,13 @@ jobs:
...
 
-      - name: Cache dependencies
-        uses: actions/cache@v2
-        with:
-          path: ./.venv
-          key: poetry-${{ inputs.dependency-hash-version }}-${{ hashFiles('poetry.lock') }}
-
       - name: Setup python and poetry and install dependencies
-        uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@v2
+        uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@TSD-1141-add-fallback-dependency-cache-keys
```
Comment (sweetb) [inline .github/workflows/python-run-checks-and-tests.yml]: Same

File: python/generate-openapi-doc/action.yml [hunk 4]
```diff
@@ -31,18 +31,13 @@ runs:
...
 
-    - name: Cache dependencies
-      uses: actions/cache@v2
-      with:
-        path: ./.venv
-        key: poetry-${{ inputs.dependency-hash-version }}-${{ hashFiles('poetry.lock') }}
-
     - name: Setup python and poetry and install dependencies
-      uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@v2
+      uses: triaxtec/github-actions/python/setup-python-and-poetry-and-install-dependencies@TSD-1141-add-fallback-dependency-cache-keys
```
Comment (sweetb) [inline python/generate-openapi-doc/action.yml]: Same

File: python/setup-python-and-poetry-and-install-dependencies/action.yml [hunk 5]
```diff
@@ -37,6 +41,15 @@ runs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ inputs.python-version }}
+
+    - name: Cache dependencies
+      uses: triaxtec/github-actions/generic/cache-dependencies@TSD-1141-add-fallback-dependency-cache-keys
```
Comment (sweetb) [inline python/setup-python-and-poetry-and-install-dependencies/action.yml]: Same

**Review by sweetb — COMMENTED** [review]
Question about hard-coded branch name

File: .github/workflows/python-deploy-serverless.yml (hunk 1 above)
Comment (emann) [reply .github/workflows/python-deploy-serverless.yml]: Goddamnit, I always forget to change these lol. Fixed.

**Review by sweetb — APPROVED** [review]
Good to go

**Pull request merged**